server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourcePhase.java

```diff
@@ -52,25 +53,23 @@ public void setNextReader(LeafReaderContext readerContext) {
 
             @Override
             public void process(HitContext hitContext) {
-                hitExecute(index, fetchSourceContext, hitContext);
+                if (SourceFieldMapper.isEnabled(fetchContext.getQueryShardContext()::getFieldType) == false) {
```

Comment:
I think we could use `QueryShardContext#isSourceEnabled` here instead of introducing a new method.

Reply:
++